plugins/chat/lib/chat/onebox_handler.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Chat
+  class OneboxHandler
+    def self.handle(url, route)
+      if route[:message_id].present?
+        message = Chat::Message.find_by(id: route[:message_id])
+        return if !message
+
+        chat_channel = message.chat_channel
+        user = message.user
+        return if !chat_channel || !user
+      else
+        chat_channel = Chat::Channel.find_by(id: route[:channel_id])
+        return if !chat_channel
+      end
```

Comment:
That's probably more a product question, but I wonder if we should try to have a 404 onebox. I will open a discussion about this.